thrift/tchannel.thrift

```diff
@@ -0,0 +1,8 @@
+struct ClientLibrary {
```

Comment:
Was thinking about that - if one Channel is registering on multiple Hyperbahn services, meta is registered on each of the service names (and the user may want custom health checks for each of those).

This seems more like global TChannel state that only needs to be registered once. However, it doesn't hurt to put this in meta, so happy to move it there if that's preferred.

E.g. it's the difference between exposing these endpoints (if it's part of Meta):
```
s1 Meta::health
s1 Meta::clientLibrary
s2 Meta::health
s2 Meta::clientLIbrary
tchannel Meta::health
tchannel Meta::clientLibrary
```
vs exposing these endpoints
```
s1 Meta::health
s2 Meta::health
tchannel Meta::clientLibrary
```